lib/http-bridge/test/integration/Test/Integration/Scenario/CLISpec.hs

```diff
@@ -0,0 +1,129 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Integration.Scenario.CLISpec
+    ( specNoCluster
+    , specWithCluster
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Types
+    ( ApiWallet )
+import Control.Monad
+    ( forM_ )
+import Data.List
+    ( length )
+import Data.Text
+    ( Text )
+import System.Exit
+    ( ExitCode (..))
+import System.Command
+    ( Exit (..), Stdout (..), command )
+import Test.Hspec
+    ( SpecWith, describe, it )
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe, shouldContain, shouldNotContain )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , expectResponseCode
+    , getFromResponse
+    , json
+    , request
+    , walletId
+    )
+
+import qualified Data.Text as T
+import qualified Network.HTTP.Types.Status as HTTP
+
+specNoCluster :: SpecWith ()
+specNoCluster = do
+
+    it "CLI - Shows help on bad argument" $  do
+        (Exit c, Stdout out) <- command [] "cardano-wallet" ["--bad arg"]
+        out `shouldContain` "Cardano Wallet CLI"
+        c `shouldBe` ExitFailure 1
+
+    it "CLI - Shows help with --help" $  do
+        (Exit c, Stdout out) <- command [] "cardano-wallet" ["--help"]
+        out `shouldContain` "Cardano Wallet CLI"
+        c `shouldBe` ExitFailure 1
```

Comment:
I wonder if this is expected...

Reply:
No it's not. I noticed some days ago (and fixed but never pushed it actually :|) while I was trying to generate code-completion for the CLI. That's not right -> Let's have 0 here and a fix :)